core/modules/parser/SelectParser.cc

```diff
@@ -237,6 +237,50 @@ class Antlr2Parser : public AntlrParser {
 };
 
 
+class Antlr4ErrorStrategy : public antlr4::DefaultErrorStrategy {
+public:
+    Antlr4ErrorStrategy(std::string const& statement) : _statement(statement) {}
```

Comment:
A few problems with this class contract as per the LSST Style Guide:
- no explicitly declkared default constructor
- no copy constructor
- no assignment operator
- the existing constructor with only parameter must be declared as `explicit`

Here is the minimal interface for the class:
```
class Antlr4ErrorStrategy : public antlr4::DefaultErrorStrategy {
public:
    explicit Antlr4ErrorStrategy(std::string const& statement);
    Antlr4ErrorStrategy() = delete;
    Antlr4ErrorStrategy(Antlr4ErrorStrategy const&) = delete;
    Antlr4ErrorStrategy& operator=(Antlr4ErrorStrategy const&) = delete;
```

This should make the class much safer by 1) avoiding its use in the implicit conversions (hence `explicit`). 2) avoiding accidental construction with uninitialized state (via the default constructor), 3) avoiding potentially dangerous (if you have a state which can't be trivially copied) by the copy ctor or the assignment operator.

See:
https://developer.lsst.io/cpp/style.html#constructors-taking-one-argument-must-be-declared-as-explicit